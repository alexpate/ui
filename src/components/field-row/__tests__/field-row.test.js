/* @flow */
/* eslint-env jest */
import React from 'react';
import {shallow} from 'enzyme';
import serializer from 'enzyme-to-json/serializer';

import FieldRow from '../field-row';

expect.addSnapshotSerializer(serializer);

describe('components/field-row', () => {
  describe('FieldRow (default)', () => {
    const create = (spread = {}) => {
      const props = {
        // (insert your default props here)
        ...spread,
      };
      const element = <FieldRow {...props} />;
      return {props, element};
    };

    it('should shallow render as expected', () => {
      const {element} = create({
        // (insert test specific props here)
        children: 'Child',
      });
      const result = shallow(element);
      expect(result).toMatchSnapshot();
    });
  });
});
