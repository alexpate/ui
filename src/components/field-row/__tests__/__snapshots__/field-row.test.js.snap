// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render shallow component ok 1`] = `
<div
  className="e1q9f2xp0 css-10v4ujk"
  style={Object {}}
>
  <div
    className="e1q9f2xp0 css-1werqj1"
    onClick={undefined}
    style={Object {}}
  >
    <div
      className="e1q9f2xp0 css-r48vgy"
      style={Object {}}
    >
      <div
        className="e1q9f2xp0 css-1o95wo4"
        style={Object {}}
      >
        <label
          className="em7k1990 css-fyza2f"
          color="navy600"
          htmlFor={undefined}
          onClick={undefined}
          size="small"
          style={Object {}}
          title={undefined}
        >
          Project name
        </label>
      </div>
      <div
        className="e6fqpax1 css-1o5tdq2"
        style={Object {}}
      >
        <input
          className="css-1yu0r67 e6fqpax2"
          defaultValue={undefined}
          disabled={false}
          name="My first project"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="My first project"
          readOnly={false}
          size="medium"
          style={
            Object {
              "width": undefined,
            }
          }
          value={undefined}
        />
      </div>
    </div>
  </div>
  <div
    className="e1q9f2xp0 css-1werqj1"
    onClick={undefined}
    style={Object {}}
  >
    <div
      className="e1q9f2xp0 css-r48vgy"
      style={Object {}}
    >
      <div
        className="e1q9f2xp0 css-1o95wo4"
        style={Object {}}
      >
        <label
          className="em7k1990 css-fyza2f"
          color="navy600"
          htmlFor={undefined}
          onClick={undefined}
          size="small"
          style={Object {}}
          title={undefined}
        >
          Task name
        </label>
      </div>
      <div
        className="e6fqpax1 css-1o5tdq2"
        style={Object {}}
      >
        <input
          className="css-1yu0r67 e6fqpax2"
          defaultValue={undefined}
          disabled={false}
          name="My first task"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="My first task"
          readOnly={false}
          size="medium"
          style={
            Object {
              "width": undefined,
            }
          }
          value={undefined}
        />
      </div>
    </div>
  </div>
</div>
`;
